src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
 export interface Figure {
-
+  color: string;
+  shape: string;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape = 'triangle';
 
-}
+  constructor(
+    public color: string,
+    private a: number,
+    private b:number,
+    private c: number,
+  ) {
+    if (
+      a < 0
+      || b < 0
+      || c < 0
+    ) {
+      throw new Error('all values should be positive');
+    }
+
+    if (
+      a + b <= c
+      || a + c <= b
+      || b + c <= a
+    ) {
+      throw new Error('Impossible to create triangle from that values');
+    }
```

Comment:
All three classes have similar checks. Maybe, let's define a private method that does this check for us ?

```typescript
private checkSides() {
  // write your code here
}
```